src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
 
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (elementData.length == size) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     public T peek() {
-        return null;
+        empty();
+        return elementData[size - 1];
     }
 
     public T pop() {
-        return null;
+        empty();
```

Comment:
as you are calling peek() on line 28 you don't need this empty(). And you can remove empty at all jest check size in peek and throw exception

Reply:
super! thank's